File: src/main/java/xyz/realmath/RationalArrays.java
package xyz.realmath;

import static com.google.common.base.Preconditions.checkArgument;

public final class RationalArrays {

  public static Rational[] base(int length, int idx) {
    checkArgument(idx >= 0, "idx < 0");
    checkArgument(idx < length, "idx >= length");
    Rational[] rationals = new Rational[length];
    for (int i = 0; i < length; i++) {
      if (i == idx) {
        rationals[i] = Rational.ONE;
      } else {
        rationals[i] = Rational.ZERO;
      }
    }
    return rationals;
  }
}

File: src/main/kotlin/xyz/realmath/RationalArrays.kt
package xyz.realmath

object RationalArrays {

    fun base(length: Int, idx: Int): Array<Rational> {
        check(idx >= 0) { "idx < 0" }
        check(idx < length) { "idx >= length" }
        return Array(length, { i ->
            if (i == idx) {
                Rational.ONE
            } else {
                Rational.ZERO
            }
        })
    }
}
